lib/auth/grpcserver.go

```diff
@@ -1772,10 +1772,22 @@ func addMFADeviceRegisterChallenge(gctx *grpcContext, stream proto.AuthService_A
 	// RPCs.
 	webIdentity := wanlib.WithInMemorySessionData(auth.Identity)
 
+	devType := initReq.DeviceType
+	if devType == proto.DeviceType_DEVICE_TYPE_UNSPECIFIED {
+		// Try and convert from legacy type.
+		// Keep conversion until 9.x, when the field is marked for deletion.
+		m := map[proto.AddMFADeviceRequestInit_LegacyDeviceType]proto.DeviceType{
```

Comment:
I would mark this with `DELETE IN` so we can easily grep for it.

Reply:
Once you delete the proto enum this one falls as a consequence, so I didn't mark it.